File: src/app/models/operation.ts
export interface Operation {
    id: number;
    type: string;
    amount: number;
    date: Date;
    description: string;
    value: number;
    status: string;
    currency: string;
    market: string;
    fee: number;
    traderName: string;
    quantity: number;
    price: number;
    category: string;
    isActive: boolean;
    priority: number;
}

export interface ChangeValueOperation {
    id: number;
    value: number;
}

File: SignalRWebSocketServer/models/Operation.cs
namespace SignalRWebSocketServer.Models
{
    public class Operation
    {
        public int Id { get; set; }
        public string Type { get; set; }
        public decimal Amount { get; set; }
        public DateTime Date { get; set; }
        public string Description { get; set; }
        public decimal Value { get; set; }
        public string Status { get; set; }
        public string Currency { get; set; }
        public string Market { get; set; }
        public decimal Fee { get; set; }
        public string TraderName { get; set; }
        public int Quantity { get; set; }
        public decimal Price { get; set; }
        public string Category { get; set; }
        public bool IsActive { get; set; }
        public int Priority { get; set; }

        public Operation(int id, string type, decimal amount, DateTime date, string description, decimal value)
        {
            Id = id;
            Type = type;
            Amount = amount;
            Date = date;
            Description = description;
            Value = value;
            Status = "Pending";
            Currency = "USD";
            Market = "Default";
            Fee = 0;
            TraderName = "Unknown";
            Quantity = 1;
            Price = 0;
            Category = "General";
            IsActive = true;
            Priority = 1;
        }
    }
}